Softeq.NetKit.Chat.Web/Controllers/DirectMessageController.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using EnsureThat;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Softeq.NetKit.Chat.Domain.Services.DomainServices;
+using Softeq.NetKit.Chat.Domain.TransportModels.Request.DirectChannel;
+using Softeq.NetKit.Chat.Domain.TransportModels.Response.DirectMessage;
+using Softeq.NetKit.Chat.SignalR.Sockets;
+using Softeq.NetKit.Chat.Web.TransportModels.Request.DirectChannel;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UpdateDirectMessageRequest = Softeq.NetKit.Chat.Web.TransportModels.Request.DirectChannel.UpdateDirectMessageRequest;
+
+namespace Softeq.NetKit.Chat.Web.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/channel")]
+    [Authorize(Roles = "Admin, User")]
+    [ApiVersion("1.0")]
+    public class DirectMessageController : BaseApiController
+    {
+        private readonly IDirectMessageSocketService _directMessageSocketService;
+        private readonly IDirectChannelService _directChannelService;
+
+        public DirectMessageController(IDirectMessageSocketService directMessageSocketService, IDirectChannelService directChannelService)
+        {
+            Ensure.That(directMessageSocketService).IsNotNull();
+            Ensure.That(directChannelService).IsNotNull();
+
+            _directMessageSocketService = directMessageSocketService;
+            _directChannelService = directChannelService;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(DirectMessageResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> AddDirectMessageAsync([FromBody] AddDirectMessageRequest request)
+        {
+            var addDirectMessageRequest = new CreateDirectMessageRequest(GetCurrentSaasUserId(), request.DirectChannelId, request.Body);
+            var directMessageResponse = await _directMessageSocketService.AddMessage(addDirectMessageRequest);
+
+            return Ok(directMessageResponse);
+        }
+
+        [HttpPut]
+        [ProducesResponseType(typeof(DirectMessageResponse), StatusCodes.Status200OK)]
+        [Route("{channelId:guid}")]
+        public async Task<IActionResult> UpdateDirectMessageAsync(Guid channelId, [FromBody] UpdateDirectMessageRequest request)
+        {
+            var updateDirectMessageRequest = new Domain.TransportModels.Request.DirectChannel.UpdateDirectMessageRequest(GetCurrentSaasUserId(),
```

Comment:
can we use using Domain.TransportModels.Request.DirectChannel

Reply:
No, because we have "using UpdateDirectMessageRequest = Softeq.NetKit.Chat.Web.TransportModels.Request.DirectChannel.UpdateDirectMessageRequest;" for Web-controller